src/config.rs

```diff
@@ -393,23 +393,103 @@ impl Config {
             Ok(())
         }
 
-        const ID_HEADER_PATTERN: &str = r"(?m)^\s*id\s*:[\sA-Za-z\d]*$";
+        /// returns true when the next occurrence of id header needs to be ignored
+        /// for cases when the section-line refers to a component definition
+        /// or if the section is a youtube type component
+        /// since user can put multiple instances of this component
+        /// in various places in the package referring to the same youtube video
+        /// having same id so ignoring these ids.
+        ///
+        /// in any other case returns false
+        fn ignore_next_id(section_line: &str) -> bool {
+            // Strip out initial '-- ' or '--- '
+            let section_line = section_line.trim_start_matches('-').trim_start();
+
+            let before_caption = section_line.split_once(':').map(|s| s.0);
```

Comment:
Move it also in `fpm::constants`, `COLON = ":"` and use it here.

Reply:
moved this to `ftd::constants::identifier` as `KV_SEPERATOR`